File: Blind75/Balanced Brackets/Balanced_Brackets.py
def isBalancedBrackets(str):
    stack = []
    pairs = {
        ')': '(',
        ']': '[',
        '}': '{'
    }
    
    for char in str:
        if char in '([{':
            stack.append(char)
            continue
        
        if char in ')]}':
            if stack.pop() != pairs[char]:
                return False
            
    return len(stack) == 0
    
inputs = ["[]", "([)]", "([]){}"]   
for input in inputs:
    print(isBalancedBrackets(input))

File: Blind75/1. Balanced Brackets/Balanced_Brackets.js
function isBalancedBrackets(str) {
  const stack = [];
  const pairs = {
    ")": "(",
    "]": "[",
    "}": "{",
  };

  for (const char of str) {
    if (char == "(" || char == "[" || char == "{") {
      stack.push(char);
      continue;
    }

    if (char == ")" || char == "]" || char == "}") {
      if (stack.pop() != pairs[char]) {
        return false;
      }
    }
  }
  
  return stack.length == 0;
}

const inputs = ["[]", "([)]", "([]){}"];
for (const input of inputs) {
  console.log(isBalancedBrackets(input));
}
